### PR TITLE
refactor: import useOutsideClick to rainbow-modules-hooks

### PR DESCRIPTION
**Pull request by yvmunayev**
fix: #1628 

## Changes proposed in this PR:
- refactor: import useOutsideClick to rainbow-modules-hooks
- fix: fix a bug in InternalOverlay

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

**Comment by LeandroTorresSicilia** [comment]
@yvmunayev take a look at CI since all unit tests are failing

File: src/components/InternalOverlay/index.js [hunk 1]
```diff
@@ -95,9 +95,10 @@ const InternalOverlay = props => {
         if (isVisible && contentMeta) {
             onOpened();
             disableBodyScroll(undefined, { reserveScrollBarGap: true });
-        } else {
-            enableBodyScroll();
         }
+        return () => {
+            enableBodyScroll();
+        };
```
Comment (LeandroTorresSicilia) [inline src/components/InternalOverlay/index.js:101]: I have doubts, whats happen when set isVisible to false and the component is not unmounted? then enableBodyScroll will not be called when it should do.

Comment (yvmunayev) [reply src/components/InternalOverlay/index.js:101]: React also sanitizes the effects of previous renderings before running the effects of the current render. So when you change isVisible enableBodyScroll is always called

https://es.reactjs.org/docs/hooks-effect.html#example-using-hooks-1

File: src/components/MultiSelect/index.js [hunk 2]
```diff
@@ -147,20 +141,19 @@ const MultiSelect = React.forwardRef((props, ref) => {
...
-        handleOutsideClick,
+        event => {
+            if (
+                event.target !== triggerRef.current.buttonRef.current &&
+                !triggerRef.current.buttonRef.current.contains(event.target)
+            ) {
+                setIsOpen(false);
+            }
+        },
+        isOpen,
```
Comment (LeandroTorresSicilia) [inline src/components/MultiSelect/index.js:154]: Multiselect outside click is not working, we should add integration test in order to avoid this happening

**Pull request merged**